### PR TITLE
Suppress shell cmd

### PR DESCRIPTION
**Pull request by thtliife**
_closes issue #131_
Small change to allow the use of a global or per-item option of suppressShellCmd for shell commands.
Adds the ability to suppress the actual command output.
### Suppress all shell commands via default config option:
_install.conf.yaml_:
```yaml
- defaults:
    shell:
      stdout: true
      stderr: true
      suppressShellCmd: true
- shell:
  - [echo Command 1, Displaying the first command]
  - [echo Command 2, Displaying the second command]
  - [echo Command 3, Displaying the third command]
  -
    command: echo Hello World
    description: Saying Hi!
  -
    command: echo Goodbye World
    description: Saying Bye!
```
will output like:
defaults-suppressshellcmd

### Suppress specific shell commands via item-specific extended config option:
_install.conf.yaml_:
```yaml
- defaults:
    shell:
      stdout: true
      stderr: true
- shell:
  - [echo Command 1, Displaying the first command]
  - [echo Command 2, Displaying the second command]
  - [echo Command 3, Displaying the third command]
  -
    command: echo Hello World
    description: Saying Hi!
    suppressShellCmd: true
  -
    command: echo Goodbye World    
    description: Saying Bye!
```
will output like:
item-suppressshellcmd

### Suppress all shell commands via default config option, with missing descriptions:
_install.conf.yaml_:
```yaml
- defaults:
    shell:
      stdout: true
      stderr: true
      suppressShellCmd: true
- shell:
  - [echo Command 1, Displaying the first command]
  - [echo Command 2]
  - [echo Command 3, Displaying the third command]
  -
    command: echo Hello World
    description: Saying Hi!
  -
    command: echo Goodbye World
```
will output like:
defaults-suppressshellcmd-without-description

### CODE REVIEW

**Comment by anishathalye** [comment]
Ah, sorry for the delay in response, it's been a busy month.

I like the idea for the feature! We'll merge it in soon.

Some feedback:

* Could we name the key something shorter? Like `quiet` (`suppressShellCmd` seems a bit verbose)
* Could you add a unit test for this (along the lines of the rest of the tests [here](https://github.com/anishathalye/dotbot/tree/master/test))? If it's too much work, you can skip it, and I can add a test, it just might take me some time to get to it.

**Comment by thtliife** [comment]
Hi @anishathalye
Sorry for my delay now too >.< I've also been busy. 
I'll get the change to rename the property to quiet, and make a unit test done tonight. :)

**Comment by Twanislas** [comment]
+1 would love this :)

**Comment by thtliife** [comment]
Have changed the option to quiet now... just working on getting the tests to run in vagrant, and I will push a unit test...

**Comment by anishathalye** [comment]
Hey -- any updates? Lmk if you need any help with our weird test infrastructure 😛

**Comment by JesusMtnez** [comment]
Any news on this? @thtliife @anishathalye

File: plugins/shell.py [hunk 1]
```diff
@@ -18,16 +18,20 @@ def handle(self, directive, data):
 
     def _process_commands(self, data):
         success = True
+        quiet_cmd = False
         defaults = self._context.defaults().get('shell', {})
         with open(os.devnull, 'w') as devnull:
             for item in data:
                 stdin = stdout = stderr = devnull
+                quiet_cmd = quiet_cmd = False
```
Comment (anishathalye) [inline plugins/shell.py]: just `quiet_cmd = False`

**Comment by anishathalye** [comment]
Slightly edited the patch, squashed, added a test, and merged. Thank you for the contribution!

**Pull request closed without merging**